recipes/gcc/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanException, ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.29.1"
+
+
+class GccConan(ConanFile):
+    name = "gcc"
+    description = "The GNU Compiler Collection includes front ends for C, " \
+                  "C++, Objective-C, Fortran, Ada, Go, and D, as well as " \
+                  "libraries for these languages (libstdc++,...). "
+    topics = ("gcc", "gnu", "compiler", "c", "c++")
+    homepage = "https://gcc.gnu.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-3.0-only"
+    settings = "os", "compiler", "arch", "build_type"
```

Comment:
Why not just:
`settings = "os", "arch"`
?

Reply:
We follow this recommendation: https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#settings, sometimes it is useful to know `self.settings.compiler` inside the recipe, so better to keep the setting and remove it from packageID.